### PR TITLE
Initial MQTT document with added bridge commands

### PR DESCRIPTION
**Pull request by bc-martin-hubacek**

### CODE REVIEW

File: source/tutorial/mqtt.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+`docker exec hub mosquitto_pub -t nodes/bridge/0/relay/i2c0-3b/set -m '{"state": true}'`
+
+== Mosquitto configuration
+
+You can edit Mosquitto settings in the configuration file which is located in `/etc/mosquitto/mosquitto.conf` inside the Docker.
+
+=== MQTT Bridge
+
+Bridging two MQTT servers together is useful for example when you have your local MQTT server behind NAT and you would like to publish messages from the outside/Internet. You have to create public MQTT server (for example on AWS server) which you bridge with your home MQTT server. In this use case your local server will connect to your public server in AWS. So you have to configure your local server with commands below.
```
Comment (jirutka) [inline source/tutorial/mqtt.adoc]: Please follow the rule “sentence per line”. By doing so, you isolate changes to the line of content that they affect, much like changing a line of code. It prevents unnecessary conflicts when merging changes, makes diffs more readable and reviews on GitHub more clear

File: source/tutorial/mqtt.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+```
+# Bridge to AWS
+connection aws
+address your.server.com
+
+# Optional password, you can also use TLS to protect credentials
```
Comment (jirutka) [inline source/tutorial/mqtt.adoc]: You just _can_? You definitely _should_ use TLS and password, this must be highly recommended.

File: source/tutorial/mqtt.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+remote_username root
+remote_password root
+
+#topic <topic> [[[out | in | both] qos-level] local-prefix remote-prefix]
+topic # in 0 /home/# /home_remote/#
+```
```
Comment (jirutka) [inline source/tutorial/mqtt.adoc]: Please always put single blank line after a block.

**Comment by bc-martin-hubacek** [comment]
Thanks for modifications. Is there a document that has the rules for formatting similar to the BigClown coding standard?
The initial MQTT tutorial was written a month ago in Markdown. Yesterday I had time to publish and I had to learn AsciiDoc and make quick changes.

**Comment by jirutka** [comment]
> Is there a document that has the rules for formatting similar to the BigClown coding standard?

Not yet, but it’s on my TODO list.

File: source/tutorial/mqtt.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
+= MQTT - Message transfer protocol
+
+== MQTT Basics
+
+- Simple, Lightweight and open message protocol
+- Each message consists of **topic** and **payload**
+- Topic is describing the meaning of the data in payload. Topic uses tree-like structure with "/" symbol. Example topics: "bedroom/temperature", "kitchen/light/set"
+- MQTT server is called **Broker** and clients can **publish** messages and **subscribe** to different types of topics
+- In subscribe topics the wildcard "+" can be used for single level subscription and symbol "&num;" for multi level subscribtion. This way you can subscribe to all temperature topics with "&num;/temperature". http://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices[More on topics]
```
Comment (bc-martin-hubacek) [inline source/tutorial/mqtt.adoc]: I used https://asciidoclive.com for output check and the hash character was missing in some places.

**Comment by jirutka** [comment]
> Thanks for modifications.

I haven’t modified anything, just commented. ;)

**Comment by bc-martin-hubacek** [comment]
Fine, this is also my first git pull. I do low level embedded firmware so I have to learn a few new things ;)

File: source/tutorial/mqtt.adoc (hunk 4 above)
Comment (jirutka) [inline source/tutorial/mqtt.adoc]: Hm, you’re right, Asciidoctor sometimes confuses it with [Custom styling](http://asciidoctor.org/docs/user-manual/#custom-styling-with-attributes). And to be even worse, escaping with `\` doesn’t work very consistently now.

So HTML entity `&num;` is fine here. Another possible workaround is to define an attribute for this, e.g. `:hash:` (under the title) and then use it as `"{hash}/temperature"`.

Why you don’t use live preview provided by Middleman (see [Writer’s setup](https://github.com/bigclownlabs/bc-doc#writers-setup))?

File: source/tutorial/mqtt.adoc [hunk 5]
```diff
@@ -3,43 +3,49 @@
 == MQTT Basics
 
 - Simple, Lightweight and open message protocol
-- Each message consists of **topic** and **payload**
-- Topic is describing the meaning of the data in payload. Topic uses tree-like structure with "/" symbol. Example topics: "bedroom/temperature", "kitchen/light/set"
-- MQTT server is called **Broker** and clients can **publish** messages and **subscribe** to different types of topics
-- In subscribe topics the wildcard "+" can be used for single level subscription and symbol "&num;" for multi level subscribtion. This way you can subscribe to all temperature topics with "&num;/temperature". http://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices[More on topics]
+- Each message consists of *topic* and *payload*
+- Topic is describing the meaning of the data in payload. Topic uses tree-like structure with “/" symbol. Example topics: `bedroom/temperature`, `kitchen/light/set`
```
Comment (jirutka) [inline source/tutorial/mqtt.adoc]: > with “/" symbol

That character after the slash is not right _double quotation mark_ ([U+201D](http://www.fileformat.info/info/unicode/char/201d/index.htm)), but _quotation mark_ ([U+0022](http://www.fileformat.info/info/unicode/char/0022/index.htm)), that is actually a symbol for inch, not proper quotation.

File: source/tutorial/mqtt.adoc (hunk 4 above)
Comment (bc-martin-hubacek) [inline source/tutorial/mqtt.adoc]: I will use Middleman in the future.

**Comment by jirutka** [comment]
> Yesterday I had time to publish and I had to learn AsciiDoc and make quick changes. 

BTW if you have more documents written Markdown, you can convert them with [kramdown-asciidoc](https://github.com/opendevise/kramdown-asciidoc). It’s in alpha state, not even released yet, but works quite well.

**Comment by bc-martin-hubacek** [comment]
I don't understand that red cross from Travis CI. Does it mean something to me? It cannot compile the doc file?

**Comment by jirutka** [comment]
That's because you're not a member of BigClownLabs organisation (I've already asked Pavel to add you) and so Travis cannot decrypt the deploy key. I've already fixed it to not fail the build in such cases, after you've opened this PR.

**Comment by bc-pavel-hubner** [comment]
I have just added Martin to organization as a member.

**Comment by bc-pavel-hubner** [comment]
Thanks, Adam. I just fixed it.

**Pull request merged**